### DISCUSSION TITLE
Pros & Cons of having multiple <launchable> tags

### DISCUSSION THREAD

**Discussion by buhtz** [ximion/appstream#594] [Q&A]
I know the specs but I do not understand them in all details: https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-launchable

It seems to be technical possible to have multiple launchable tags

    <launchable type="desktop-id">backintime-qt.desktop</launchable>
    <launchable type="desktop-id">backintime-qt-root.desktop</launchable>

The specs says it is possible but should be avoided to have multiple ones of type "desktop-id". But the specs give no reason.
Also `appstreamcli validate` do not have a problem with it. Also `gnome-software --show-metainfo` do not warn about it.

Are there good reasons against it?

My arguments in context my own project are:

I have two launchers (desktop-files) because the GUI application can be started with root-rights (via pkexec).

The two launchers are in the same package. So it wouldn't be a problem if the second launcher is not mentioned and would use `X-AppStream-Ignore=true`. But from Debian AppStream guideline I know that the name and description from appstream data is used (if available) instead of the one a desktop file provides. If this is really the case I can remove the Comments fields from the desktop file and use the appstream file for this. It makes translation managment even easier because I can ignore the desktop file.

**Comment by ximion** [comment]
Cons: Pretty much nobody and nothing supports multiple launchable tags.
Pros: ? (in this case)

To elaborate: Multiple launchable elements are intended to be used for applications like LibreOffice, or KiCAD potentially, where one application ships with distinct, individually launchable components (like Writer, Impress, Draw, etc.). When the user requests to launch that app, they should be presented a dialog to select which one of the apps they should launch.

For clarity and simplicity, I would pick the variant that the user would most commonly launch and use that as single launchable, so the user isn't confused if a software center does show a dialog on which component should be launched.

Also keep in mind that order matters here (and e.g. category information is merged from the first launchable entry when the final metadata is composed).

Ultimately it's your call, but I'd just go with the simplest option here :-)

**Reply by buhtz** [reply]
Mhm... This is still not clear enough. It is possible (LibreOffice example) but not recommended. But why not?
